### ISSUE TITLE
Android Port

### ISSUE THREAD

**Issue by HanzoDev1375** [shadps4-emu/shadPS4#490]

**Comment by SolidnyWonsz** [comment]
Your phone will melt

**Comment by HanzoDev1375** [comment]
> Your phone will melt

Seriously?

**Comment by polybiusproxy** [comment]
It certainly won't run full speed. For an Android port, a recompiler would be needed to translate the x86 code to ARM, which is a hard task. It is feasible, but the emulator is still in early stages so a long time will pass before that happens.

**Comment by HanzoDev1375** [comment]
> It certainly won't run full speed. For an Android port, a recompiler would be needed to translate the x86 code to ARM, which is a hard task. It is feasible, but the emulator is still in early stages so a long time will pass before that happens.

nice

**Comment by SRC267** [comment]
Or simply run shadps4 with those Windows emulators for android, like Winlator, Mobox etc. They are capable of running the PC version of emulators like RPCS3, CEMU, PCSX2, Ryujinx etc.. Although the performance of games is slow. It's a temporary solution for now until we wait for many years.

**Comment by Marcinoo97** [comment]
> It certainly won't run full speed. For an Android port, a recompiler would be needed to translate the x86 code to ARM, which is a hard task. It is feasible, but the emulator is still in early stages so a long time will pass before that happens.

lol. u contributed to this project yet not know everything.  I wonder how is MacOS running it so well then.

**Comment by red-prig** [comment]
In this emulator MacOS running does not work well

**Comment by Foul-Tarnished** [comment]
> > It certainly won't run full speed. For an Android port, a recompiler would be needed to translate the x86 code to ARM, which is a hard task. It is feasible, but the emulator is still in early stages so a long time will pass before that happens.
> 
> lol. u contributed to this project yet not know everything. I wonder how is MacOS running it so well then.

...
It's thanks to Rosetta2, that let ARM Apple Silicon run any x86_64 apps.

The translation is not done by shadps4 with a custom JIT recompile.
Otherwise it could be reused for an eventual Android port

**Comment by georgemoralis** [comment]
Not a priority  atm

**Comment by Degerz** [comment]
> It certainly won't run full speed. For an Android port, a recompiler would be needed to translate the x86 code to ARM, which is a hard task. It is feasible, but the emulator is still in early stages so a long time will pass before that happens.

If there were a port I imagine that the only compatible graphics hardware options out there would either be Adreno or modern Exynos GPUs. All of the necessary BC texture formats are supported on Adreno but that's about where all of its positives end and while modern Exynos designs are theoretically the closest match to the PS4's graphics hardware, its texture unit implementation does not support all of the BC formats and there's currently no way to use custom graphics drivers for them either to potentially take advantage of creating/developing custom extensions to take advantage of speed up/accuracy boosts.

**Comment by Arkuarius** [comment]
> It certainly won't run full speed. For an Android port, a recompiler would be needed to translate the x86 code to ARM, which is a hard task. It is feasible, but the emulator is still in early stages so a long time will pass before that happens.

@SolidnyWonsz This is an OLD thread but I still feel like chiming in - yes, our phones will melt, but not exactly because of the recompiler. Depending on how well the x86-64 to Arm64 recompiler is made, it could actually be really fast - Box64 on Android is the biggest proof that it's not actually that demanding, and you can even run many MODERN games like System Shock at 60 FPS on a phone, which the PS4 cannot because the game is CPU-bottlenecked to just 30 FPS on the PS4. 

Bottom line is, no, an x86 to Arm recompiler genuinely wouldn't be too demanding, but our phones would still "melt" regardless because of the GPU side.

Unlike with PC games (like the PC version of System Shock) where you can lower the graphical settings down and the resolution down to 720p and 540p, PS4 games are made specifically for the PS4 with no graphical wiggle room, and the PS4's GPU is miles better than its CPU. Emulating PS4 games would be extremely heavy on the GPU and only few phones have GPUs powerful enough to match the PS4's. If a ShadPS4 Android port WERE to happen, phones' GPUs would be locked at 100% usage at maximum clock, which would use high TDP and as a result, our phones would "melt".
